src/Utils.hs

```diff
@@ -15,5 +24,34 @@ boldText offset textPic = pictures
 numberToPicture :: Int -> Picture
 numberToPicture n = translate (-20) (-20) . scale 0.5 0.5 . boldText 1 . text $ show n
 
+
+-- Converts text to picture
 textToPicture :: String -> Picture
-textToPicture s = text $ s
\ No newline at end of file
+textToPicture s = text $ s
+
+
+-- Asigna un color a cada valor de celda
+cellColor :: Int -> Color
+cellColor value = case value of
+    2    -> makeColorI 238 228 218 255   -- beige
+    4    -> makeColorI 237 224 200 255   -- light orange
+    8    -> makeColorI 242 177 121 255   -- orange
+    16   -> makeColorI 245 149 99 255    -- light red
+    32   -> makeColorI 246 124 95 255    -- red
+    64   -> makeColorI 246 94 59 255     -- strong red
+    128  -> makeColorI 237 207 114 255   -- light yellow
+    256  -> makeColorI 237 204 97 255    -- yellow
+    512  -> makeColorI 237 200 80 255    -- strong yellow
+    1024 -> makeColorI 237 197 63 255    -- light green
+    2048 -> makeColorI 237 194 46 255    -- green
+    _    -> makeColorI 205 193 180 255   -- default color (para cualquier otro valor)
+
+
+    -- Function to create a new empty board
+createEmptyBoard :: Int -> Board
+createEmptyBoard a = array indexRange $ zip (range indexRange) (cycle [Empty])
+  where
+    indexRange = ((0, 0), (a - 1, a - 1))
+
+getEmptyCells :: Board -> [(Int, Int)]
+getEmptyCells board = filter (\coord -> board ! coord == Empty) (indices board)
```

Comment:
• No instance for ‘Eq Cell’ arising from a use of ‘==’
• In the expression: board ! coord == Empty
  In the first argument of ‘filter’, namely
    ‘(\ coord -> board ! coord == Empty)’
  In the expression:
    filter (\ coord -> board ! coord == Empty) (indices board)typecheck(-Wdeferred-type-errors)